app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -184,161 +181,217 @@ public boolean contactExists(final String name) throws IOException {
+        final String databasePassJson = gsonParser.toJson(databasePass);
+        LOGGER.info("Database pass: " + databasePassJson);
+
+        final Writer writer = this.fileSystem.getWriter(passwordPath);
+
+        writer.write(databasePassJson);
+        writer.close();
+        LOGGER.info("Created the password file: %s", this.passwordPath);
+
     }
 
+
     /**
-     * Initialize the specified section in the database. This does nothing
-     * if the specified section of the database already exists.
+     * Gets the databasePass from the databaseMap
+     *
+     * @throws IOException throws IOException if the database isn't loaded yet.
      */
-    private void initializeDatabase() throws IOException {
-        final File file = new File(this.profilesPath);
-        if (!file.exists()) {
-            LOGGER.warn("Part of the database didn't exist: %s", this.profilesPath);
-            final Writer writer = this.fileSystem.getWriter(this.profilesPath);
-            writer.write("[]");
-            writer.close();
-            LOGGER.info("Created the part of the database: %s", this.profilesPath);
+    private DatabasePass getDatabasePass() throws IOException {
+        final Object object = databaseMap.get(DATABASE_PASS);
+        if (object instanceof DatabasePass) {
+            return (DatabasePass) object;
+        } else {
+            throw new IOException(DATABASE_NOT_CREATED);
         }
     }
 
     /**
-     * Deserialize the instance using readObject to ensure invariants and security.
-     * @param stream The serialized object to be deserialized
+     * Gets the database from the databaseMap
+     *
+     * @throws IOException throws IOException if the database isn't loaded yet.
      */
-    private void readObject(final ObjectInputStream stream) throws IOException, ClassNotFoundException {
-        stream.defaultReadObject();
-        ensureClassInvariant();
+    private Database getDatabase() throws IOException {
+        final Object object = databaseMap.get(DATABASE);
+        if (object instanceof Database) {
+            return (Database) object;
+        } else {
+            throw new IOException(DATABASE_NOT_CREATED);
+        }
     }
 
     /**
-     * Ensure that the instance meets its class invariant
-     * @throws InvalidObjectException Thrown when the state of the class is unstable
+     * Gets the encryption key from the databaseMap
+     *
+     * @throws IOException throws IOException if the database isn't loaded yet.
      */
-    private void ensureClassInvariant() throws InvalidObjectException {
-        assertNotNull(this.contactsPath);
-        assertNotNull(this.profilesPath);
-        assertNotNull(this.fileSystem);
+    private SecretKey getEncryptionKey() throws IOException {
+        final Object object = databaseMap.get(ENCRYPTION_KEY);
+        if (object instanceof SecretKey) {
+            return (SecretKey) object;
+        } else {
+            throw new IOException(DATABASE_NOT_CREATED);
```

Comment:
Can you create a new class extending RuntimeException in the package exceptions? IOException is too vague (actually, it has little to do with IO at all)